### ISSUE TITLE
Wildcard definition for class in global namespace matches any namespace

### ISSUE THREAD

**Issue by Micr0mega** [PHP-DI/PHP-DI#861]
The project I am working on has quite some classes defined in the global namespace. 
When adding a wildcard definition for these classes, not only those classes are matched, but also matching class names in any namespace.
This is quite unexpected, since the documentation states:
> the wildcard does not match across namespaces

For example:

```
return [
    'Job*' => DI\autowire()->method('foo', 'bar'),
];
```
will call `foo('bar')` on class `JobDoSomething`, but also on class `SomeNamespace\JobScheduler`, which is not in the global namespace.

I am unfortunately on version 6.3.5, because PHP 7.2 ... But the code for matching the wildcard (regex) looks to be equal in both 6.4 and master.

**Comment by mnapoli** [comment]
That's an interesting bug indeed, happy to merge a PR with a test that reproduces it + a fix!

**Comment by Micr0mega** [comment]
I'd be happy to, but would you be willing to release 6.3.6 to allow this fix to work on PHP 7.2? And then probably 6.4.1 as well for PHP 7.4.

If so, should I branch from master and target my PR there, so you can backport the fix to 6.x? Or would you suggest another approach?
